aws/chef-bootstrap.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+# Code-dot-org Chef bootstrap script.
+# One-liner to run:
+# aws s3 cp s3://cdo-dist/chef/bootstrap.sh - | sudo bash -s -- [options]
```

Comment:
We expect to find the latest version, which should be as stable and backwards-compatible as possible. This is modeled after the Chef omnitruck install script at https://omnitruck.chef.io/install.sh, which behaves in a similar fashion to provide a portable entrypoint for installing Chef- (see discussion in [this blog post](https://www.chef.io/blog/2015/07/16/5-ways-to-deal-with-the-install-sh-curl-pipe-bash-problem/)). If this is not sufficient or if we expect breaking changes to happen often to this script it should be possible to add branch-specific variants to parallel the Chef-cookbook provisioning process (or commit-specific versions if we want be even more fine-grained with it).